### ISSUE TITLE
stb_ds crashes when inserting struct

### ISSUE THREAD

**Issue by srdjanstipic** [nothings/stb#992]
The following program crashes:

```c
#define STB_DS_IMPLEMENTATION
#include "stb_ds.h"

int main() {
  struct { char *key; int value;} *t = NULL, s = {"a", 1};
  sh_new_strdup(t);
  shputs(t, s);
  shfree(t);
  return 0;
}
```

**Comment by nothings** [comment]
This is a known limitation that is missing from the documentation.  The "struct" form of the string hash APIs can't be used with the arena/strdup modes.

(The issue is that internally the hash table attempts to strdup() the key field, but then the final update operation for struct operations is to overwrite the entire struct with the passed-in structure.)

I'll look into having it assert if it's used incorrectly.

**Comment by makuto** [comment]
This issue should be closed after PR #993, right? Or was there a reason it is kept open?

**Comment by srdjanstipic** [comment]
Should be closed.
